util/linuxfw/nftables_runner.go

```diff
@@ -804,6 +804,32 @@ func (n *nftablesRunner) AddChains() error {
 	return n.conn.Flush()
 }
 
+// createDummyPostroutingChains is used to create dummy postrouting chains in
+// netfilter via nftables, as a last resort measure to detect that nftables can
+// be used. It cleans up the dummy chains after creation.
+func (n *nftablesRunner) createDummyPostroutingChains() error {
+	const chainName = "ts-test-postrouting"
+	polAccept := nftables.ChainPolicyAccept
+	for _, table := range n.getNATTables() {
+		// Create the nat table if it doesn't exist, this table name is the same
+		// as the name used by iptables-nft and ufw. We install rules into the
+		// same conventional table so that `accept` verdicts from our jump
+		// chains are conclusive.
+		nat, err := createTableIfNotExist(n.conn, table.Proto, "nat")
+		if err != nil {
+			return fmt.Errorf("create nat table: %w", err)
+		}
+		table.Nat = nat
+		if err = createChainIfNotExist(n.conn, chainInfo{nat, chainName, nftables.ChainTypeNAT, nftables.ChainHookPostrouting, nftables.ChainPriorityNATSource, &polAccept}); err != nil {
+			return fmt.Errorf("create %q chain: %w", chainName, err)
+		}
+		if err := deleteChainIfExists(n.conn, nat, chainName); err != nil {
```

Comment:
shouldn't you start by doing a deleteChainIfExists before the create, so we can actually for sure test a create?

Otherwise on entry to the function if the chain already existed (e.g. interrupted at unfortunate spot previously), this will do nothing?

Reply:
`createChainIfNotExist` ensures that the chain exists and has the appropriate values, and if thats the case we don't need to recreate it.